app/src/main/java/com/example/cyclofit/ui/fragment/KcalCombinedChartFragment.kt

```diff
@@ -0,0 +1,183 @@
+
+        //This is for accessing the user data form Firestore collection
+        mFireStore.collection(Constants.USERS)
+            // the document id to get the field of User.
+            .document(FirebaseAuth.getInstance().currentUser!!.uid)
+            .get().addOnSuccessListener {
+                userWeight = it.toObject(User::class.java)!!.weight.toDouble().toString()
+                if (json != null) {
+                    sp = gson.fromJson(json, type)
+                }
+                val combinedData = CombinedData().apply {
+                    setData(generateLineData())
+                    setData(generateBarData())
+                    setValueTypeface(Typeface.DEFAULT)
+                }
+                setCombinedChart(combinedData)
+            }
+        return binding.root
+    }
+
+    private fun getCaloriesForLineChart(userWeight: String): MutableList<Entry> {
+        val barArrayList = mutableListOf<Entry>()
+        for ((x, i) in sp.withIndex()) {
+            val y = i.time.toInt()
+            val met = 12 // for bicycles
+            val wt = userWeight.toFloat() // in kg
+            var totalCalsBurnt: Float = y * (met * 3.5f * wt)
+            totalCalsBurnt /= 200 * 1000
+            barArrayList.add(BarEntry(x.toFloat(), totalCalsBurnt))
+        }
+        return barArrayList
+    }
+
+    private fun getCaloriesForBarChart(userWeight: String): MutableList<BarEntry> {
+        val barArrayList = mutableListOf<BarEntry>()
+        for ((x, i) in sp.withIndex()) {
+            val y = i.time.toInt()
+            val met = 12 // for bicycles
+            val wt = userWeight.toFloat() // in kg
+            var totalCalsBurnt: Float = y * (met * 3.5f * wt)
+            totalCalsBurnt /= 200 * 1000
+            barArrayList.add(BarEntry(x.toFloat(), totalCalsBurnt))
+        }
+        return barArrayList
+    }
+
+    private fun setCombinedChart(combinedData: CombinedData) {
+        binding.combinedChart.apply {
+            data = combinedData
+            invalidate()
+            description.isEnabled = false
+            setBackgroundColor(Color.TRANSPARENT)
+            setDrawGridBackground(false)
+            setDrawBarShadow(false)
+            isHighlightFullBarEnabled = false
+            // draw bars behind lines
+            drawOrder = arrayOf(
+                DrawOrder.BAR, DrawOrder.BUBBLE, DrawOrder.CANDLE, DrawOrder.LINE, DrawOrder.SCATTER
+            )
+            legend.apply {
+                isWordWrapEnabled = true
+                verticalAlignment = Legend.LegendVerticalAlignment.BOTTOM
+                horizontalAlignment = Legend.LegendHorizontalAlignment.CENTER
+                orientation = Legend.LegendOrientation.HORIZONTAL
+                setDrawInside(false)
+            }
+            axisRight.apply {
+                setDrawGridLines(false)
+                setDrawAxisLine(false)
+                setDrawLabels(false)
+                axisMinimum = 0f
```

Comment:
we are using these again from line 130. It will be better to put this inside of a function.  Also make sure that you put this kind of refactoring throughout your code